weave

```diff
@@ -205,6 +205,40 @@ dns_arg_count() {
     fi
 }
 
+collect_rewrite_hosts_args() {
+    REWRITE_HOSTS_ARG_COUNT=0
+    case "$1" in
+    --rewrite-hosts)
+        REWRITE_HOSTS=1
+        REWRITE_HOSTS_ARG_COUNT=1
+        ;;
```

Comment:
> huh? why do we have a `--rewrite-hosts` all of a sudden?

Because I think it is more user-friendly: it does not make sense to recognize `--no-rewrite-hosts` but to fail with a `usage` when he/she uses `--rewrite-hosts`... I think we are doing the same with `--with-dns`/`--without-dns`

> Also, it would be simpler to inline this function at the call site.

Probably, but I think `collect_rewrite_hosts_args` creates a pattern where we do `collect_*_args` and `shift *_COUNT`, and we can use it in more parts of the code...